src/Stryker.Core/Stryker.Core/StrykerRunner.cs

```diff
@@ -134,5 +134,28 @@ private void AnalyseCoverage(StrykerOptions options)
                 }
             }
         }
+
+        /// <summary>
+        /// In the case of multiple projects we wrap them inside a wrapper root component. Otherwise the only project root will be the root component.
+        /// </summary>
+        /// <param name="projectComponents">A list of all project root components</param>
+        /// <param name="options">The current stryker options</param>
+        /// <returns>The root folder component</returns>
+        private CsharpFolderComposite AddRootFolderIfMultiProject(IEnumerable<IProjectComponent> projectComponents, StrykerOptions options)
+        {
+            var rootComponent = new CsharpFolderComposite
```

Comment:
It would be better to have a new type for this (RootFolderComposite?) because CsharpFolderComposite is specific to csharp, which is not needed at this stage in the process.

Reply:
Why do we even have a CsharpFolder? A folder should not care what type of project it is in right? It can just contain files. The files are indeed language specific.

Reply:
Because we use the syntaxtrees on folder level. For example, when compiling we take all syntaxtrees from the root component. The syntaxtree type is csharp specific, so the folder must be as well.

Reply:
We also use the folder composite to add compilation only syntax trees to the project (for example the stryker helpers)